File: src/test/java/integration/status/GameStatusTestConfiguration.java
package integration.status;

import org.mockito.Mockito;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;

import com.matag.game.event.EventSender;
import com.matag.game.security.SecurityHelper;

@Configuration
@ComponentScan("com.matag.game.status")
public class GameStatusTestConfiguration {

  @Bean
  @Primary
  public SecurityHelper securityHelper() {
    return Mockito.mock(SecurityHelper.class);
  }

  @Bean
  @Primary
  public EventSender eventSender() {
    return Mockito.mock(EventSender.class);
  }

}

File: src/test/kotlin/integration/status/GameStatusTestConfiguration.kt
package integration.status

import com.matag.game.event.EventSender
import com.matag.game.security.SecurityHelper
import org.mockito.Mockito
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary

@Configuration
@ComponentScan("com.matag.game.status")
open class GameStatusTestConfiguration {
    @Bean
    @Primary
    open fun securityHelper(): SecurityHelper? {
        return Mockito.mock<SecurityHelper?>(SecurityHelper::class.java)
    }

    @Bean
    @Primary
    open fun eventSender(): EventSender? {
        return Mockito.mock<EventSender?>(EventSender::class.java)
    }
}